Scripts/demand/freight.py

```diff
@@ -15,17 +15,34 @@ def calc_freight_traffic(self, mode):
         production_forecast = self.generate_trips(zone_data_forecast, mode)
         # Remove zero values
         base_mtx = self.base_demand.get_data(mode).clip(0.000001, None) 
-        external_growth = 1.1
-        production_calib = base_mtx.sum(1)
-        production = self.calibrate(production_calib[:len(production_base)], 
+        production = self.calibrate(base_mtx.sum(1), #[:len(production_base)], 
                                     production_base, 
                                     production_forecast)
-        # External demand is generated for the last centroids
-        external_base = production_calib[len(production):]
-        external_production = external_growth * external_base
-        production = numpy.append(production, external_production)
-        # Add harbour traffic!
-        demand = self.fratar(production, base_mtx)
+        # # External demand is generated for the last centroids
+        # external_base = production_calib[len(production):]
+        # external_growth = 1.1
+        # external_production = external_growth * external_base
+        # production = numpy.append(production, external_production)
+        zone_numbers = self.base_demand.get_zone_numbers()
+        mtx = pandas.DataFrame(base_mtx, zone_numbers, zone_numbers)
+        prod = pandas.Series(production, zone_numbers)
+        # If forecast>5*base, destination choice is replaced by area average
+        cond = production_forecast < 5*production_base
+        for i in xrange(0, 30):
```

Comment:
It's not obvious where does the numbers 30 and 5 come from, perhaps use constant names?

SOME_INDEX_TO_STOP=30

Reply:
Made the code more flexible, allowing arbitrary zone numbers.